File: 2025/minimumOperationsToMakeArrayValuesEqualToK.ts
function minOperations(nums: number[], k: number): number {
    let sorted = nums.sort((a, b) => a-b);
    if (sorted[0] < k) return -1;
    let operations = 0;

    for (let i = sorted.length-1; i > -1; i--) {
        if (sorted[i] < sorted[i+1]){
            operations++;
        }
    }

    if (sorted[0] > k) operations++;

    return operations;
};

console.log(minOperations([5,2,5,4,5], 2)); // 2
console.log(minOperations([2,1,2], 2)); // -1
console.log(minOperations([9,7,5,3], 4)); // 2

File: 2025/go/easy/minimumOperationsToMakeArrayvaluesEqualToK.go
package easy

import "sort"

func minOperations(nums []int, k int) int {
	sort.Ints(nums)
	if nums[0] < k {
		return -1
	}
	operations := 0
	if nums[0] > k {
		operations++
	}
	for i, _ := range nums {
		if i+1 < len(nums) {
			if nums[i] < nums[i+1] {
				operations++
			}
		}
	}

	return operations
}
